pkg/kubectl/cmd/get.go

```diff
@@ -296,26 +296,26 @@ func RunGet(f *cmdutil.Factory, out io.Writer, errOut io.Writer, cmd *cobra.Comm
 			allErrs = append(allErrs, err)
 		}
 
-		// the outermost object will be converted to the output-version, but inner
-		// objects can use their mappings
-		version, err := cmdutil.OutputVersion(cmd, clientConfig.GroupVersion)
-		if err != nil {
-			return err
-		}
 		res := ""
 		if len(infos) > 0 {
 			res = infos[0].ResourceMapping().Resource
 		}
 
-		obj, err := resource.AsVersionedObject(infos, !singular, version, f.JSONEncoder())
```

Comment:
Each versionedObject would have a different `reflect.Type` than [generic `infos.Object` in HumanReadablePrinter](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/get.go#L380), so when filtering I would have something like `*api.Pod` [as the map key for the pod filter](https://github.com/kubernetes/kubernetes/pull/31163/commits/5de8f7ed7f5a9502666d6a6008c432587bb3f19a#diff-baa0ea36296411cb41a034c9c5f6ec13R44), which would not match with the `reflect.Type` of versioned objects.

A solution that would allow me to keep versionedObjects here would be to have two keys (one for versioned, another for unversioned) in the [filterMap](https://github.com/kubernetes/kubernetes/pull/31163/commits/5de8f7ed7f5a9502666d6a6008c432587bb3f19a#diff-baa0ea36296411cb41a034c9c5f6ec13R28) pointing to the same filterFunc for that resource. WDYT?